packages/next-swc/crates/next-api/src/middleware.rs

```diff
@@ -225,9 +225,9 @@ pub(crate) async fn get_paths_from_root(
         .map({
             move |&file| async move {
                 let path = &*file.ident().path().await?;
-                let relative = root
-                    .get_path_to(path)
-                    .context("file path must be inside the root")?;
+                let Some(relative) = root.get_path_to(path) else {
```

Comment:
As discussed with @sokra this error doesn't make sense, we want to skip files that aren't matching for this handling.